docs/CUSTOM-BRANCHES.md

```diff
@@ -0,0 +1,80 @@
+Some custom branches are very helpful for gtp2ogs, but they are not implemented 
+in the official gtp2ogs because it would add extra maintainance to maintain them
+
+To use them, just replace your current gtp2ogs files with the branche's file, 
+as explained in the gtp2ogs full tutorial, except that you have to download the 
+custom branch ZIP instead of the devel branch : 
+- for linux : [3A3) Recommended : Upgrade gtp2ogs.js from old branch to “devel” branch (latest)](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3A3-linux-optional-upgrade-to-devel.md)
+- for windows : [3B3) Recommended : Upgrade gtp2ogs from old branch to devel (latest) branch](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3B3-windows-optional-upgrade-to-devel.md)
+
+Below are the most notable custom branches freely available : 
+
+## Dorus/dorus branch :
+
+This branch has 2 main features : 
+
+- 1) it adds the ability to display PV (variations) of Leela Zero and PhoenixGo
+
+Here is an example of game using it for PhoenixGo (meta-金毛测试-20b) : 
+https://online-go.com/game/16510570
+
+Here is an example of game using it for LeelaZero-ELF (NightlyLeela) : 
+https://online-go.com/game/16097813
+
+Here is an example of game using it for LeelaZero-15b (15bTurboLeela) : 
+https://online-go.com/game/16511596
+
+To use it, you have to add the gtp2ogs argument `--logogspv LZ` if your bot is 
+leela zero or a leela zero variant, or `--logogspv PG` if you bot is the native 
+PhoenixGo
+
+- 2) This branch also comes with "gears" settings to play faster when 
+winrate of your leela zero bot (or variants) is above X% winrate
+
+This is very useful to keep lost games to needlessly consume a lot of time
+
+This is how it works : 
+
+- gear 0 : no need to do anything, it's the default specified bot speed 
+(seconds/move)
+
+- gear 1 (max 10 seconds per move at X% winrate) : if you add the gtp2ogs 
+argument `--fast1 X`, for example `--fast1 90` it means the bot will play 
+at maximum 10 seconds per move if winrate is higher than 90% in that example
+
+- gear 2 (max 1 second per move at X% winrate) : if you add the gtp2ogs 
+argument `--fast2 X`, for example `--fast2 95` it means the bot will play 
+at maximum 1 second per move if winrate is higher than 95% in that example
+
+it is possible to combine both gears and have the bot play, in that example, 
+using `--fast1 90 --fast2 95` :
+- at gear 0 (default speed, no need to add any argument) if winrate is <90% 
+- at gear 1 (10s per move) if winrate is >90% but <95%
+- at gear 2 (1s per move) if winrate is >= 95%
+
+You also need to install split2 to use this branch 
+(as admin on windows, as sudo on linux) :
+
+```npm install -g split2```
```

Comment:
I think just `npm install` install will work since i fixed up packages.json

Reply:
indeed but you never know, so i put it like that for now

Reply:
You do know. Also installing it global is rediculous.

Reply:
i dont mind doing it if you're 100% sure there wont be any exception, 

and if we are sure about, then shouldnt we also delete this from gtp2ogs readme ? 

https://github.com/online-go/gtp2ogs#3-optional--install-any-missing-nodejs-packages

> 3. Optional : install any missing node.js packages
> This step can be skipped unless you have issues or bugs

> you may need to install extra tools if the Most common usage below fails, such as (run as sudo > for linux, and as admin for windows)

> npm install -g socket.io-client
> npm install -g optimist
> npm install -g tracer

Reply:
Yes, as long as they are present in packages.json,you only need to type `npm install`.

To do so you can use `npm install xxx - -save-prod`

Reply:
ok, so `npm install gtp2ogs - save-prod` ? the weird space is at the right place ? (im becoming spaces fanatic too :rofl: )

Reply:
1) No i fixed the space.
2) This is only necessarily to update `packages.json`, once that's done and checked in, others can run just `npm install`.

Reply:
ok, all clear

so the gtp2ogs readme will also serve your branch since you share the same readme, all the more reason to fix it !

new readme : 

> in case it fails, run `npm install gtp2ogs - save-prod` to update packages to install list in packages.json
then run : `npm install` to install all missing packages

correct @Dorus ?

if i'm understanding right, then its a nice and very efficient fix to the readme (and to all custom branches at the same time)

Reply:
no no no no no just no.

Only devs should run `--save-prod` to updat `package.json`. Users only need to call `npm install`

Reply:
ok got it, gonna commit it !

Reply:
fixed in https://github.com/online-go/gtp2ogs/pull/129/commits/02db84a9669deadf2041a3e52e2e62df9b0b593a